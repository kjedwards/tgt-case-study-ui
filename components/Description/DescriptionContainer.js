/* ----------------------------------------------
 * Container for the product information
 *
 * Author: Kaylee Edwards (z0027hd)
 * Last Modified: 06/08/2018
 * Last Modified By: Kaylee Edwards (z0027hd)
 * --------------------------------------------*/

import React, {Component} from 'react';
import styled, {css} from 'styled-components'

import { SIZES } from '../../constants/DeviceWidth'

import Price from './Price'
import Promotions from './Promotions'
import Quantity from './Quantity'
import BuyButtonContainer from './BuyButtonContainer'
import Returns from './Returns'
import ShareContainer from './ShareContainer'
import ProductHighlights from './ProductHighlights'

const DescContainer = styled.div`
  order: 2;
  display: flex;
  flex-direction: column;
  max-width: 300px;

  @media(max-width: ${SIZES.TABLET}) {
    order: 1;
  }
`


class DescriptionContainer extends Component {
    render () {
      return(
        <DescContainer>
          <Price offer={this.props.offer}></Price>
          <Promotions promos={this.props.promo}></Promotions>
          <Quantity></Quantity>
          <BuyButtonContainer channelCode={this.props.channelCode}></BuyButtonContainer>
          <Returns></Returns>
          <ShareContainer></ShareContainer>
          <ProductHighlights highlights={this.props.highlights}></ProductHighlights>
        </DescContainer>
      )
    }
}


export default DescriptionContainer
