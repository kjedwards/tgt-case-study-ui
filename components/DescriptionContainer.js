import React, {Component} from 'react';
import styled, {css} from 'styled-components'

import { sizes } from '../constants/DeviceWidth'

const DescContainer = styled.div`
  background-color: blue;
  height: 800px;
  width: 50%;
  order: 2;

  @media(max-width: ${sizes.mobileLarge}) {
    order: 1;
  }
`


class DescriptionContainer extends Component {
    render () {
      return(
        <DescContainer></DescContainer>
      )
    }
}


export default DescriptionContainer
