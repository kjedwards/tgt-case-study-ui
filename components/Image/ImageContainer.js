import React, {Component} from 'react';
import styled, {css} from 'styled-components'

import ProductTitle from './ProductTitle'
import ProductImage from './ProductImage'
import ProductCarousel from './ProductCarousel'

import { sizes } from '../../constants/DeviceWidth'

const ImgContainer = styled.div`
  background-color: red;
  width: 40%;
  height: 500px;
  min-width: 400px;
  order: 0;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;

  @media(max-width: ${sizes.laptop}) {
      max-width: 400px;
  }

  @media(max-width: ${sizes.desktop}) {
      max-width: 600px;
  }
`


class ImageContainer extends Component {
    render () {
      return(
        <ImgContainer>
          <ProductTitle></ProductTitle>
          <ProductImage></ProductImage>
          <ProductCarousel></ProductCarousel>
        </ImgContainer>
      )
    }
}


export default ImageContainer

//export default connect(mapStateToProps)(App)
