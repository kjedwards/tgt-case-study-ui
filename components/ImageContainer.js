import React, {Component} from 'react';
import styled, {css} from 'styled-components'

import ProductTitle from './ProductTitle'
import ProductImage from './ProductImage'
import ProductCarousel from './ProductCarousel'

const ImgContainer = styled.div`
  background-color: red;
  height: 500px;
  width: 40%;
  min-width: 400px;
  order: 0;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
`


class ImageContainer extends Component {
    render () {
      return(
        <ImgContainer>
          <ProductTitle></ProductTitle>
          <ProductImage></ProductImage>
          <ProductCarousel></ProductCarousel>
        </ImgContainer>
      )
    }
}


export default ImageContainer

//export default connect(mapStateToProps)(App)
