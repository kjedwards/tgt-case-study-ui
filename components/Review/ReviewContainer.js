import React, {Component} from 'react';
import styled, {css} from 'styled-components'

import { sizes } from '../../constants/DeviceWidth'

import ReviewTop from './ReviewTop'
import ReviewSummary from './ReviewSummary'

const RvwContainer = styled.div`
  background-color: green;
  width: 40%;
  min-width: 400px;
  order: 1;
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;

  @media(max-width: ${sizes.tablet}) {
    order: 2;
  }
`


class ReviewContainer extends Component {
    render () {
      return(
        <RvwContainer>
          <ReviewTop></ReviewTop>
          <ReviewSummary></ReviewSummary>
        </RvwContainer>
      )
    }
}


export default ReviewContainer
