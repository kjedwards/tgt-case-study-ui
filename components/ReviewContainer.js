import React, {Component} from 'react';
import styled, {css} from 'styled-components'

import { sizes } from '../constants/DeviceWidth'

const RvwContainer = styled.div`
  background-color: green;
  height: 200px;
  width: 40%;
  min-width: 400px;
  order: 1;

  @media(max-width: ${sizes.mobileLarge}) {
    order: 2;
  }
`


class ReviewContainer extends Component {
    render () {
      return(
        <RvwContainer></RvwContainer>
      )
    }
}


export default ReviewContainer
