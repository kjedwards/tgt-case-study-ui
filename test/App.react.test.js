import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

//Imported Components
import DescriptionContainer from '../components/Description/DescriptionContainer'
import ImageContainer from '../components/Image/ImageContainer'
import ReviewContainer from '../components/Review/ReviewContainer'

//Description Mocks
jest.mock('../components/Description/BuyButtonContainer', () => 'buy-button-container')
jest.mock('../components/Description/Price', () => 'price')
jest.mock('../components/Description/ProductHighlights', () => 'prouct-highlights')
jest.mock('../components/Description/Promotions', () => 'promotions')
jest.mock('../components/Description/Quantity', () => 'quantity')
jest.mock('../components/Description/Returns', () => 'returns')
jest.mock('../components/Description/ShareContainer', () => 'share-container')

//Image Mocks
jest.mock('../components/Image/ProductCarousel', () => 'product-carousel')
jest.mock('../components/Image/ProductImage', () => 'product-image')
jest.mock('../components/Image/ProductTitle', () => 'product-title')

//Review Mocks
jest.mock('../components/Review/ReviewSummary', () => 'review-summary')
jest.mock('../components/Review/ReviewTop', () => 'review-top')


describe('DescriptionContainer Component', () => {
  it('RENDERER snapshot', () => {
    const tree = renderer.create(<DescriptionContainer />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('ENZYME snapshot', () => {
    const tree = mount(<DescriptionContainer />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})

describe('ImageContainer Component', () => {
  it('RENDERER snapshot', () => {
    const tree = renderer.create(<ImageContainer />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('ENZYME snapshot', () => {
    const tree = mount(<ImageContainer />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})

describe('ReviewContainer Component', () => {
  it('RENDERER snapshot', () => {
    const tree = renderer.create(<ReviewContainer />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('ENZYME snapshot', () => {
    const tree = mount(<ReviewContainer />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})
