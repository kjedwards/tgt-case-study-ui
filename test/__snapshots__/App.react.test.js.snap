// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescriptionContainer Component ENZYME snapshot 1`] = `
<DescriptionContainer>
  <styled.div>
    <div
      className="sc-bdVaJa fdsMyH"
    >
      <price />
      <promotions />
      <quantity />
      <buy-button-container />
      <returns />
      <share-container />
      <prouct-highlights />
    </div>
  </styled.div>
</DescriptionContainer>
`;

exports[`DescriptionContainer Component RENDERER snapshot 1`] = `
<div
  className="sc-bdVaJa fdsMyH"
>
  <price />
  <promotions />
  <quantity />
  <buy-button-container />
  <returns />
  <share-container />
  <prouct-highlights />
</div>
`;

exports[`ImageContainer Component ENZYME snapshot 1`] = `
<ImageContainer>
  <styled.div>
    <div
      className="sc-bwzfXH dPgevf"
    >
      <product-title />
      <product-image />
      <product-carousel />
    </div>
  </styled.div>
</ImageContainer>
`;

exports[`ImageContainer Component RENDERER snapshot 1`] = `
<div
  className="sc-bwzfXH dPgevf"
>
  <product-title />
  <product-image />
  <product-carousel />
</div>
`;

exports[`ReviewContainer Component ENZYME snapshot 1`] = `
<ReviewContainer>
  <styled.div>
    <div
      className="sc-htpNat kFdBUn"
    >
      <review-top />
      <review-summary />
    </div>
  </styled.div>
</ReviewContainer>
`;

exports[`ReviewContainer Component RENDERER snapshot 1`] = `
<div
  className="sc-htpNat kFdBUn"
>
  <review-top />
  <review-summary />
</div>
`;
